tests/network/test_add_intercept.py

```diff
@@ -255,3 +257,87 @@ async def test_add_intercept_type_pattern_port_empty_invalid(websocket):
         "error": "invalid argument",
         "message": "URL pattern must specify a port"
     } == exception_info.value.args[0]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "url_patterns",
+    [
+        [
+            {
+                "type": "string",
+                "pattern": "https://www.example.com/",
+            },
+        ],
+        # [
+        #     {
+        #         "type": "pattern",
+        #         "protocol": "https",
+        #         "hostname": "www.example.com",
+        #         "pathname": "/",
+        #     },
+        # ],
+        # [
+        #     {
+        #         "type": "string",
+        #         "pattern": "https://www.example.com/",
+        #     },
+        #     {
+        #         "type": "pattern",
+        #         "protocol": "https",
+        #         "hostname": "www.example.com",
+        #         "pathname": "/",
+        #     },
+        # ],
+    ],
+    ids=[
+        "string",
+        #  "pattern",
+        #  "string and pattern",
+    ])
+async def test_add_intercept_blocks(websocket, context_id, url_patterns):
+    url = "https://www.example.com/"
+
+    await subscribe(websocket, ["cdp.Fetch.requestPaused"])
+
+    result = await execute_command(
+        websocket, {
+            "method": "network.addIntercept",
+            "params": {
+                "phases": ["beforeRequestSent"],
+                "urlPatterns": url_patterns,
+            },
+        })
+
+    assert result == {
+        "intercept": ANY_UUID,
+    }
+
+    await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": url,
+                "context": context_id,
+            }
+        })
+
+    event_response = await wait_for_event(websocket, "cdp.Fetch.requestPaused")
```

Comment:
do we want to test using a bidi event? or alternatively add a todo.

Reply:
There's no BiDi event equivalent to `Fetch.requestPaused`.

And the other events (response started, response completed, etc) require other intercept commands/methods which aren't yet available.

I am considering the TODO to write more tests, but IMHO this test is OK as is - having at least one test that tests the CDP event should be fine. That said, we can add a TODO to stop using example.com